frontend/src/components/layout/Navigation.tsx

```diff
@@ -36,14 +57,35 @@ export const Navigation = () => {
     </a>
   );
 
-  const signInButtonRef = useGaViewPing({
+  const signInFxaFlowTracker = useFxaFlowTracker({
     category: "Sign In",
     label: "nav-profile-sign-in",
+    entrypoint: "relay-sign-in-header",
   });
+  // document is undefined when prerendering the website,
+  // so just use the production URL there:
+  const signInUrl = new URL(
+    getRuntimeConfig().fxaLoginUrl,
+    typeof document !== "undefined"
+      ? document.location.origin
+      : "https://relay.firefox.com"
+  );
+  signInUrl.searchParams.append("form_type", "button");
+  signInUrl.searchParams.append("entrypoint", "relay-sign-in-header");
+  if (signInFxaFlowTracker.flowData) {
+    signInUrl.searchParams.append(
+      "flowId",
+      signInFxaFlowTracker.flowData.flowId
+    );
+    signInUrl.searchParams.append(
+      "flowBeginTime",
+      signInFxaFlowTracker.flowData.flowBeginTime
+    );
+  }
```

Comment:
suggestion (blocking): this code is all duplicated from the `relay-sign-up-heaader` code above? Consolidate into a function that can be called from each spot.

Reply:
bf76ad70

I also added an API mock so we can emulate the behaviour, which actually helped me find a bug (I'd used `flowBeginType` rather than `flowBeginTime` - I've probably been corrupted by TS), so we can now actually see the parameters being added [here](https://deploy-preview-1880--fx-relay-demo.netlify.app/).